### PR TITLE
chore!: convert addon to v2 format

### PR DESCRIPTION
**Pull request by SergeAstapov**
This is breaking change as it now requires the consuming addon or the add to have ember-auto-import >= v2.

This follows https://github.com/embroider-build/embroider/blob/main/PORTING-ADDONS-TO-V2.md#part-4-convert-addon-to-v2 guide

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -12,7 +12,16 @@
     "lint:test-app": "yarn workspace test-app run lint",
     "lint:fix": "npm-run-all --aggregate-output --continue-on-error --parallel lint:fix:*",
     "lint:fix:addon": "yarn workspace ember-cookies run lint:fix",
-    "lint:fix:test-app": "yarn workspace test-app run lint:fix"
+    "lint:fix:test-app": "yarn workspace test-app run lint:fix",
+    "prepare": "yarn workspace ember-cookies run build",
```
Comment (SergeAstapov) [inline package.json:16]: this is needed to build the addon as v2 addon needs to be prebuilt before consumption

File: packages/ember-cookies/.babelrc.js [hunk 2]
```diff
@@ -1,9 +0,0 @@
-const { buildEmberPlugins } = require('ember-cli-babel');
```
Comment (SergeAstapov) [inline packages/ember-cookies/.babelrc.js:1]: this is not needed as we now provide `babel.config.json`

File: packages/ember-cookies/.gitignore [hunk 3]
```diff
@@ -19,6 +17,6 @@ yarn-error.log
 testem.log
 .DS_Store
 
-# ember-try
-.node_modules.ember-try/
-package.json.ember-try
+# .md copies from release
```
Comment (SergeAstapov) [inline packages/ember-cookies/.gitignore:20]: to make those files explorable on github, we keep them in the root folder.

to make those files packed at publish time, we copy them to the addon folder.
This allows to keep single source of truth and at the same time keep all the nice parts of DX

File: packages/ember-cookies/package.json [hunk 4]
```diff
@@ -5,30 +5,51 @@
...
     "test": "tests"
   },
-  "repository": "https://github.com/simplabs/ember-cookies",
+  "files": [
+    "addon-main.js",
+    "dist"
+  ],
   "scripts": {
+    "build": "rollup --config",
+    "copyDoc": "cp ../../*.md .",
```
Comment (SergeAstapov) [inline packages/ember-cookies/package.json:27]: this is where copy of the files happens.

File: packages/ember-cookies/src/clear-all-cookies.js [hunk 5]
```diff
@@ -1,10 +1,11 @@
 import { assert } from '@ember/debug';
```
Comment (SergeAstapov) [inline packages/ember-cookies/src/clear-all-cookies.js:1]: moved the file to new location to keep existing import path `ember-cookies/clear-all-cookies` work

File: packages/ember-cookies/src/test-support/index.js [hunk 6]
```diff
@@ -0,0 +1 @@
+export { default as clearAllCookies } from '../clear-all-cookies';
```
Comment (SergeAstapov) [inline packages/ember-cookies/src/test-support/index.js:1]: as this is actually test helper, by conventions it should come `ember-cookies/test-support` path.

@marcoow please let me know if you don't agree.

**Comment by SergeAstapov** [comment]
@marcoow as a followup and to ease release process, I can open PR to introduce `release-it` similar to https://github.com/lifeart/ember-click-outside-modifier/pull/16/files

this really simplifies release management for v2 addons as otherwise it might be error prune as it needs to consist of at least:
* bumping version in `packages/ember-cookies`
* updating `ember-cookies` version in `packages/test-app`
* updating changelog (not sure how it's managed so far, seem manual step?)
* creating GitHub release (not sure how it's managed so far, seem manual step?)

File: packages/ember-cookies/.gitignore [hunk 7]
```diff
@@ -19,6 +17,6 @@ yarn-error.log
...
+/CHANGELOG.md
+/README.md
```
Comment (marcoow) [inline packages/ember-cookies/.gitignore:22]: newline missing

File: packages/ember-cookies/babel.config.json [hunk 8]
```diff
@@ -0,0 +1,7 @@
+{
+  "plugins": [
+    "@embroider/addon-dev/template-colocation-plugin",
+    ["@babel/plugin-proposal-decorators", { "legacy": true }],
+    "@babel/plugin-proposal-class-properties"
+  ]
+}
```
Comment (marcoow) [inline packages/ember-cookies/babel.config.json:7]: :eyes

File: packages/ember-cookies/src/test-support/index.js (hunk 6 above)
Comment (marcoow) [reply packages/ember-cookies/src/test-support/index.js:1]: Yes, if this is the convention let's change the way this needs to be imported as this is breaking anyway :+1

Comment (SergeAstapov) [reply packages/ember-cookies/src/test-support/index.js]: done! import path outlined in the `README.md`

**Review by marcoow — APPROVED** [review]

**Pull request merged**